CMakeLists.txt

```diff
@@ -443,6 +431,39 @@ else()
 endif()
 list(APPEND SRC_FILES ${CU_SOURCES})
 
+if (ENABLE_MSCCL)
+  add_definitions(-DCOMPILE_MSCCL)
+  set(MSCCL_SOURCES
+    src/collectives/device/msccl_kernel_impl.h
+    src/include/msccl/msccl_kernel.h
+    src/include/msccl/msccl_lifecycle.h
+    src/include/msccl/msccl_parser.h
+    src/include/msccl/msccl_scheduler.h
+    src/include/msccl/msccl_setup.h
+    src/include/msccl/msccl_status.h
+    src/include/msccl/msccl_struct.h
+    src/misc/msccl/msccl_lifecycle.cc
+    src/misc/msccl/msccl_parser.cc
+    src/misc/msccl/msccl_setup.cc
+    src/misc/msccl/msccl_status.cc
+    src/collectives/msccl.cc
+  )
+else()
+  set(MSCCL_SOURCES
+    src/include/msccl/msccl_lifecycle.h
+    src/include/msccl/msccl_parser.h
+    src/include/msccl/msccl_scheduler.h
+    src/include/msccl/msccl_setup.h
+    src/include/msccl/msccl_status.h
+    src/include/msccl/msccl_struct.h
+    src/misc/msccl/msccl_lifecycle.cc
+    src/misc/msccl/msccl_parser.cc
+    src/misc/msccl/msccl_setup.cc
+    src/misc/msccl/msccl_status.cc
+  )
+endif()
+list(APPEND SRC_FILES ${MSCCL_SOURCES})
+  
```

Comment:
Move the common ones back into list of all sources, and only differentiate the kernel file.